agent/engine/docker_task_engine.go

```diff
@@ -489,28 +489,41 @@ func (engine *DockerTaskEngine) handleDockerEvents(ctx context.Context) {
 // handleDockerEvent is responsible for taking an event that correlates to a
 // container and placing it in the context of the task to which that container
 // belongs.
-func (engine *DockerTaskEngine) handleDockerEvent(event DockerContainerChangeEvent) bool {
-	log.Debug("Handling a docker event", "event", event)
+func (engine *DockerTaskEngine) handleDockerEvent(event DockerContainerChangeEvent) {
+	seelog.Debugf("Handling a docker event: %s", event.String())
 
 	task, taskFound := engine.state.TaskByID(event.DockerID)
 	cont, containerFound := engine.state.ContainerByID(event.DockerID)
 	if !taskFound || !containerFound {
-		log.Debug("Event for container not managed", "dockerId", event.DockerID)
-		return false
+		seelog.Debugf("Event for container not managed, container: %s, task found: %v, container found: %v",
+			event.DockerID, taskFound, containerFound)
+		return
 	}
+
+	// Container health status change doesnot affect the container status
+	// no need to process this in task manager
+	if event.Type == api.ContainerHealthEvent {
+		if cont.Container.HealthStatusShouldBeReported() {
+			seelog.Debugf("Updating container health status: %s", event.DockerContainerMetadata.Health)
+			cont.Container.SetHealthStatus(event.DockerContainerMetadata.Health)
+			engine.saver.Save()
```

Comment:
I don't think we need to persist this in the state file. In the event of a restart, inspecting the container should be sufficient to get the health status of the container. wdyt? imo, we can delete this line.